ffi.c

```diff
@@ -526,7 +682,7 @@ static void* check_pointer(lua_State* L, int idx, struct ctype* ct)
         ct->is_unsigned = 1;
         ct->pointers = 0;
         lua_pushnil(L);
-        return (void*) (uintptr_t) lua_tonumber(L, idx);
+        return (void*) (uintptr_t) lua_tonumber(L, idx); // TODO in Lua 5.3: maybe change to lua_tointeger
```

Comment:
Yeah, I think this should be changed but it's not super urgent. In practice 64-bit pointers can fit in doubles on the architectures we care about.

It's a little annoying because, `lua_isinteger` is 5.3-only and `lua_tointegerx` is 5.2+ only